gobblin-modules/gobblin-metadata/src/main/java/gobblin/metadata/types/GlobalMetadata.java

```diff
@@ -222,7 +222,23 @@ public void setContentType(String contentType) {
    * @return
    */
   public String getContentType() {
-    return (String)getDatasetMetadata(CONTENT_TYPE_KEY);
+    return (String) getDatasetMetadata(CONTENT_TYPE_KEY);
+  }
+
+  public void setPermission(String permission) {
+    setDatasetMetadata(PERMISSION_KEY, permission);
```

Comment:
Discussed offline. Reverted changes in `GlobalMetadata`. Added a `PermissionMetadataParser` to parse `permission` related info.